src/main/java/org/jenkinsci/plugins/codescene/CodeSceneBuilder.java

```diff
@@ -376,16 +387,17 @@ public void perform(Run<?, ?> build, FilePath workspace, Launcher launcher, Task
             if (isAnalyzeBranchDiff() && getBaseRevision() != null) {
                 analyzeWorkOnBranchFor(build, workspace, launcher, listener, codesceneConfig, currentCommit, branch);
             }
-
         } catch (RemoteAnalysisException e) {
-            listener.error("Remote failure as CodeScene couldn't perform the delta analysis: %s", e);
+            String message = String.format("Remote failure as CodeScene couldn't perform the delta analysis: %s", e);
+            listener.error(message);
             // This is necessary to log the complete stacktrace - listener only shows exception message
             // Note: this log is visible only in the jenkins logs, not in the job's console log.
-            logger.log(Level.WARNING, "Remote failure as CodeScene couldn't perform the delta analysis: %s", e);
-            build.setResult(buildResultForFailedAnalysisDependsOn(letBuildPassOnFailedAnalysis));
+            logger.log(Level.WARNING, message);
+            setBuildResult(build, buildResultForFailedAnalysisDependsOn(letBuildPassOnFailedAnalysis), message);
```

Comment:
`buildResultForFailedAnalysisDependsOn` marks build as _success_ or failure depending on the value of `letBuildPassOnFailedAnalysis`. However, `setBuildResult` doesn't make such a distinction and always throws an Exception. Shouldn't it do the same thing, that is do _not_ throw an exception if `letBuildPassOnFailedAnalysis` is true?

Reply:
I modified the code so the exception will be thrown only if result is set to FAILURE. Also `DeltaAnalysis` is modified throw IOException which will always fail the build or stare in case cannot reach/connect codescene server.

the flag `letBuildPassOnFailedAnalysis` will affect only the `RemoteAnalysisException` catch branch